### PR TITLE
fix #47 STARFIT_DATA should specify an additional directory rather than a replacement

### LINKED ISSUE #47: `STARFIT_DATA` should specify an additional directory rather than a replacement
currently a directory can be set where the data lives, `STARFIT_DATA`, but maybe this should be to allow specifying an additional directory rather than a replacement?

_Originally posted by @2sn in https://github.com/conradtchan/starfit/issues/24#issuecomment-1279851577_

### PR DESCRIPTION
**Pull request by dliptai**

### CODE REVIEW

**Review by conradtchan — APPROVED** [review]
LGTM 👍

**Comment by dliptai** [comment]
@2sn if you approve, feel free to merge it in

**Comment by 2sn** [comment]
@dliptai Apologies, I don't like this change. 
* It is a good idea to put the finding in a separate subroutine, but why get rid of the `pathlib` functions?  They are now preferred over `os.path`.   
* I can see some potential issues with the code as is, because it no longer looks specifically at `db` or `ref` directories for the specific files.
* As I said, I will deal with that, this is something I can do, I understand the code now well enough.

**Comment by 2sn** [comment]
@dliptai I think I would rewrite the generic find function to
* include the specific subdirectory as parameter instead of `**` in `glob`.  
* use `pathlib` instead of `os.path` (because `os.path` is supposed to be phased out.

**Comment by 2sn** [comment]
@dliptai Your code is very generous in translating, I would have
* just used the plain actual name, as this allows to use the routine more generically in the future, and in the code people should know the directory; although I do agree that this decouples the data location specified elsewhere from the actual location.  Using plane names is more transparent, and if desired to have something more generic, I'd rather use use a variable defined in `__init__` such as `DB_PATH`
* `dtype` seems not a good choice for a name as it is used a lot in the NumPy context.

I will take care of some changes, if you don't mind?  (in a few days, busy week)

**Comment by 2sn** [comment]
@dliptai this is the changes I had in mind.  OK for you?

File: README.md [hunk 1]
```diff
@@ -150,16 +150,15 @@ Custom stellar data and model database files can always be used by providing a f
 ```
 export STARFIT_DATA='/your/custom/data'
 ```
-The custom data directory should have the same structure as `src/starfit/data`, i.e. it should contain the `db`, `ref`, and `stars` directories:
+Files found in the custom data directory will take precedence over the default data directory.
+It is suggested that your custom data directory should have the same structure as `src/starfit/data`, i.e. it should contain the `db`, `ref`, and `stars` directories:
```
Comment (dliptai) [inline README.md:154]: Just need to modify the readme to reflect the new changes

**Comment by dliptai** [comment]
Looks good! Just need to modify the readme to reflect the new changes

**Review by 2sn — APPROVED** [review]

**Pull request merged**